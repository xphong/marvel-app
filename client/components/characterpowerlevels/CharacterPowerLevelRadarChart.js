import React from 'react';
import PropTypes from 'prop-types';

import { Radar } from 'react-chartjs-2';
import WithChart from '../hocs/WithChart';

const RadarChart = WithChart(Radar);

export const CharacterPowerLevelRadarChart = props => {
  const { characterChartData } = props;

  const chartOptions = {
    scale: {
      ticks: {
        maxTicksLimit: 3,
        display: false
      }
    }
  };

  return (
    <div>
      <RadarChart data={characterChartData} chartOptions={chartOptions} />
    </div>
  );
};

CharacterPowerLevelRadarChart.propTypes = {
  characterChartData: PropTypes.object.isRequired
};

export default CharacterPowerLevelRadarChart;
