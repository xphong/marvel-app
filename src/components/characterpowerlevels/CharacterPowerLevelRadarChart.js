import React, { Component, PropTypes } from 'react';

import { Radar as RadarChart } from 'react-chartjs';

export default class CharacterPowerLevelRadarChart extends Component {
  constructor(props){
    super(props);
  }

  getCharacterChartData(character) {
    return {
      labels: ["Eating", "Drinking", "Sleeping", "Designing", "Coding", "Cycling", "Running"],
      datasets: [
        {
          label: "Character Data Set",
          data: [28, 48, 40, 19, 96, 27, 100]
        }
      ]
    };
  }

  render() {
    const {character} = this.props;
    const characterChartData = this.getCharacterChartData(character);

    return (
        <RadarChart data={characterChartData} width='600' height='400'/>
    );
  }
}

CharacterPowerLevelRadarChart.propTypes = {
  character: PropTypes.object.isRequired
};
