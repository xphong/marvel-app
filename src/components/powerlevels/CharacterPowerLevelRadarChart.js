import React, { Component, PropTypes } from 'react';

import { Radar as RadarChart } from 'react-chartjs';

export default class CharacterPowerLevelRadarChart extends Component {
  constructor(props){
    super(props);
  }

  getCharacterChartData(character) {
    return {
    labels: ["Eating", "Drinking", "Sleeping", "Designing", "Coding", "Cycling", "Running"],
    datasets: [
        {
            label: "My Second dataset",
            backgroundColor: "rgba(255,99,132,0.2)",
            borderColor: "rgba(255,99,132,1)",
            pointBackgroundColor: "rgba(255,99,132,1)",
            pointBorderColor: "#fff",
            pointHoverBackgroundColor: "#fff",
            pointHoverBorderColor: "rgba(255,99,132,1)",
            data: [28, 48, 40, 19, 96, 27, 100]
        }
    ]
    };
  }

  render() {
    const {character} = this.props;

    const characterChartData = this.getCharacterChartData(character);

    return (
        <RadarChart data={characterChartData} width='600' height='400'/>
    );
  }
}

CharacterPowerLevelRadarChart.propTypes = {
  character: PropTypes.object.isRequired
};
