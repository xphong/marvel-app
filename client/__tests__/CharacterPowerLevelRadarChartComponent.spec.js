import React from 'react';
import ReactDOM from 'react-dom';

import CharacterPowerLevelRadarChart from '../components/characterpowerlevels/CharacterPowerLevelRadarChart';

describe('CharacterPowerLevelRadarChart Component', () => {
  it('should render CharacterPowerLevelRadarChart', () => {
    const props = {
      characterChartData: {
        label: 'Radar chart of awesomeness'
      }
    };

    const wrapper = shallow(<CharacterPowerLevelRadarChart {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
