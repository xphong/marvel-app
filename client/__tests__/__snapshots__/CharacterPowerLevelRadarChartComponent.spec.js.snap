// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterPowerLevelRadarChart Component should render CharacterPowerLevelRadarChart 1`] = `
<div>
  <Component
    chartOptions={
      Object {
        "scale": Object {
          "ticks": Object {
            "display": false,
            "maxTicksLimit": 3,
          },
        },
      }
    }
    data={
      Object {
        "label": "Radar chart of awesomeness",
      }
    }
  />
</div>
`;
